internal/addrs/resourcemode_string.go

```diff
@@ -11,20 +11,26 @@ func _() {
 	_ = x[InvalidResourceMode-0]
 	_ = x[ManagedResourceMode-77]
 	_ = x[DataResourceMode-68]
+	_ = x[EphemeralResourceMode-69]
 }
 
 const (
 	_ResourceMode_name_0 = "InvalidResourceMode"
-	_ResourceMode_name_1 = "DataResourceMode"
+	_ResourceMode_name_1 = "DataResourceModeEphemeralResourceMode"
```

Comment:
I suppose this is correct _generated_ (i.e. hard to reason about) code but it still confuses me that we ended up with both modes in the same string, rather than a separate constant. 🤔

Reply:
(replied once in the wrong place, so sorry if you get a similar comment twice ;)) I'm not sure why this uses a different algorithm when we use random characters, because these are usually packed into a single constant. If these were an series of consecutive values like `ioota`, you would get `"InvalidResourceModeDataResourceModeEphemeralResourceModeManagedResourceMode"`.